pkg/identity/cache/cache.go

```diff
@@ -103,26 +102,63 @@ type identityWatcher struct {
 	stopChan chan bool
 }
 
+func collectEvent(event allocator.AllocatorEvent, added, deleted IdentityCache) bool {
+	if gi, ok := event.Key.(globalIdentity); ok {
+		id := identity.NumericIdentity(event.ID)
+		lbls := gi.LabelArray()
+		if event.Typ == kvstore.EventTypeCreate {
+			added[id] = lbls
+		} else {
+			deleted[id] = lbls
+		}
+		return true
+	}
+	log.Warningf("Ignoring unknown identity type '%s': %+v",
```

Comment:
Please use the `log.WithFields(...).Warning(...)` style across this PR. For log management systems and even our CI, this makes errors easier to track. We now do things in the CI like counting the number of warnings with a unique message to figure out the most popular problems, so if the content of the log message has variable contents it can mess with that type of operation over the logs.

Reply:
Done, mostly. For this one I did not, as this is something that should happen only if something is severely wrong with the source code, and could not readily figure out if there are suitable fields to use. Should I just invent new ones in this case?

Reply:
(Sorry, missed this earlier) @jrajahalme if you think there's a case to reuse across multiple packages, be sure to add new fields to `pkg/logging/logfields/`, otherwise I'm fine with just defining the parameter titles using direct strings here.

Given the `%+v`, I think it's important to do so with this particular message.